test/integration/quic_http_integration_test.cc

```diff
@@ -1045,5 +1078,38 @@ TEST_P(QuicInplaceLdsIntegrationTest, ReloadConfigUpdateDefaultFilterChain) {
   codec_client_1->close();
 }
 
+TEST_P(QuicInplaceLdsIntegrationTest, EnableAndDisableEarlyData) {
+  enable_quic_early_data_ = true;
+  inplaceInitialize(/*add_default_filter_chain=*/false);
+
+  auto codec_client_0 = makeRawHttp3Connection(
+      makeClientConnectionWithHost(lookupPort("http"), "www.lyft.com"), absl::nullopt,
+      /*wait_for_1rtt_key*/ true);
+  makeRequestAndWaitForResponse(*codec_client_0);
+  codec_client_0->close();
+
+  // Modify 1st transport socket factory to disable early data.
+  ConfigHelper new_config_helper(
+      version_, *api_, MessageUtil::getJsonStringFromMessageOrDie(config_helper_.bootstrap()));
+  new_config_helper.addQuicDownstreamTransportSocketConfig(/*enable_early_data=*/false);
+
+  new_config_helper.setLds("1");
+  test_server_->waitForCounterGe("listener_manager.listener_in_place_updated", 1);
+  test_server_->waitForGaugeGe("listener_manager.total_filter_chains_draining", 1);
+
+  test_server_->waitForGaugeEq("listener_manager.total_filter_chains_draining", 0);
+  // The 2nd connection should try to do 0-RTT but get rejected and QUICHE will transparently retry
+  // the request after handshake completes.
+  auto codec_client_2 = makeRawHttp3Connection(
+      makeClientConnectionWithHost(lookupPort("http"), "www.lyft.com"), absl::nullopt,
+      /*wait_for_1rtt_key*/ false);
+  makeRequestAndWaitForResponse(*codec_client_2);
+
+  EnvoyQuicClientSession* quic_session =
+      static_cast<EnvoyQuicClientSession*>(codec_client_2->connection());
+  EXPECT_FALSE(quic_session->EarlyDataAccepted());
```

Comment:
If I'm reading this right, it looks like both tests end up with early data rejected. Is there any way to get a test where we accept early data?